library/Mockery/Expectation.php

```diff
@@ -301,7 +301,12 @@ protected function _matchArg($expected, &$actual)
             return $expected->match($actual);
         }
         if ($expected instanceof \Hamcrest_Matcher) {
-            return $expected->matches($actual);
+            try {
+                assertThat($actual, is($expected));
```

Comment:
Let's use namespaced version of `assertThat`: `\Hamcrest\MatcherAssert::assertThat`.